timemachine/fe/mle.py

```diff
@@ -212,3 +214,64 @@ def infer_node_vals_and_errs(
     dg_err = bootstrap_estimates.std(0)
 
     return dg, dg_err
+
+
+def infer_node_vals_and_errs_networkx(
+    nx_graph: nx.Graph,
+    edge_diff_prop: str,
+    edge_stddev_prop: str,
+    node_val_prop: str,
+    node_stddev_prop: str,
+    ref_nodes: Sequence[Any],
+    n_bootstrap: int = 100,
+    seed: int = 0,
+) -> nx.Graph:
+    """Version of :py:func:`timemachine.fe.mle.infer_node_vals_and_errs` that accepts a networkx graph.
+
+    Parameters
+    ----------
+    nx_graph: networkx.Graph
+        Networkx graph
+    edge_diff_prop: str
+        Edge property to use for differences
+    edge_stddev_prop: str
+        Edge property to use for standard deviations
+    node_val_prop: str
+        Node property to use for absolute values
+    node_stddev_prop: str
+        Node property to use for standard deviations. If missing in reference nodes, assume zero.
+    ref_nodes: sequence
+        Reference nodes (must have properties ref_node_val_prop and ref_node_stddev_prop)
```

Comment:
Ah, thanks. Fixed in https://github.com/proteneer/timemachine/pull/987/commits/51c8097c344ad957a14845df40c912b255b1cba1

> Uncertainty: Is it preferable to overwrite `node_val_prop` on `ref_nodes` (as currently implemented), or to keep these properties separate (i.e. read some `ref_node_*_props` from `ref_nodes`, but write MLE results to different property names `node_*_props`)?

I'm inclined to overwrite in the result for simplicity and since we're returning a copy, so the original values will still be accessible. Don't have a terribly strong preference, though.

Reply:
As discussed offline, we have some use cases where we want to avoid overwriting reference properties due to potential for causing confusion. Added separate arguments for `node_*_prop` and `ref_node_*_prop` in https://github.com/proteneer/timemachine/pull/987/commits/1c02d73eb7db1b9b170d04bb64f7fcb42565b861